### PR TITLE
[AIRFLOW-3217] Airflow log view and code view should wrap

### PR DESCRIPTION
**Pull request by jmeickle**
### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues: https://issues.apache.org/jira/browse/AIRFLOW-3217

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

DAG code view before/after:
screenshot 2018-12-04 12 19 46
screenshot 2018-12-04 12 19 23

Log view before/after:
screenshot 2018-12-04 12 20 21
screenshot 2018-12-04 12 20 43

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: it's css

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by ron819** [comment]
I would love if it would be possible to choose between wrap and not wrap per log. Maybe a setting in the log page?

However in any case this PR updates only the old UI (which will be deprecated when https://github.com/apache/incubator-airflow/pull/4142 is merged) You should update also the rbac UI.

**Review by kaxil — CHANGES_REQUESTED** [review]
Better to give this option to the user, maybe config in `airflow.cfg`. Also as pointed in other comment, can you update this in FAB-UI

**Comment by jmeickle** [comment]
Hi @kaxil - I added some js and config for this. The intended behavior is:

- The new config will default to True
- If True, DAG code and TI log views will default to wrapping. If False, they will default to existing behavior.
- There is clickable text on each view that will toggle wrapping; this is not persisted.

I don't have a working local dev instance right now however, so I am hoping someone else can take a look at this.

**Comment by kaxil** [comment]
Will this cause a bad experience if a user is trying to select or copy the code. A user click will toggle the behaviour.

**Comment by jmeickle** [comment]
The PR adds link text at the top of the page to toggle rather than clicking
the code block.

A scrolling toggle widget absolute positioned to screen bottom would be
better UX, but that's a much larger project since you would then need to
account for screen height changing after press.

**Review by kaxil — APPROVED** [review]

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -330,6 +330,8 @@ cookie_secure = False
 # Set samesite policy on session cookie
 cookie_samesite =
 
+# Default setting for wrap toggle on DAG code and TI log views.
+default_wrap = True
```
Comment (kaxil) [inline airflow/config_templates/default_airflow.cfg]: Can we change the default setting to be `False`, to keep the current default behavior

Comment (jmeickle) [reply airflow/config_templates/default_airflow.cfg]: That sounds fine to me but I'm not at my dev env so I don't have the ability to commit + squash that this moment

**Comment by jmeickle** [comment]
@kaxil I changed the default setting to not wrap. I also tested this for the code view, and the button does work. I wasn't able to test with the log view, as I couldn't get a functioning worker environment to generate logs.

One complication here with the code view is that the line number code creates a 2x1 table: all the line numbers are in one cell, and all the code is in another cell. This means that there is no way to actually wrap the line numbers correctly to the line they represent. Fixing that would require tweaking how we generate the HTML from pygments, to create one table row per line of code. But that's getting into a larger PR, and I'd rather just drop code view wrapping from this PR rather than implementing that right now...

**Comment by jmeickle** [comment]
```
            html_code = highlight(
                code, lexers.PythonLexer(), HtmlFormatter(linenos="inline"))
```

Switching to this in `views.py` would result in wrap-compatible line numbers for the code view but also break copying and pasting. It doesn't look like pygments has an out of the box solution that does both.

**Comment by jmeickle** [comment]
@kaxil Do you think this is mergeable at this point? If additional work is required to make the code view acceptable; then I would rather just go ahead with the log view changes, and hope that someone picks up code view in the future. I rarely look at code view, but log view wrapping is a problem for me and my team every day.

**Comment by kaxil** [comment]
cc @feng-tao - who knows more about the UI

**Review by ashb — APPROVED** [review]

**Pull request merged**